README.md

```diff
@@ -0,0 +1,188 @@
+
+Now it's time for a mini-stress test, by running a [defined set of actions](https://github.com/cescoffier/spring-to-quarkus-demo/blob/main/justfile#L15). Before every run, make sure you have an empty database, by running `just restart-infra`. In a new window, at the top-level of the repository, run
+
+```
+just stress
+```
+
+Collect the startup time and footprint after the stress actions complete. 
+You can read the startup time from the application log. It will be something like 
+
+```
+2024-06-27T11:56:59.475+01:00  INFO 37043 --- [           main] m.escoffier.spring.todo.TodoApplication  : Started TodoApplication in 2.779 seconds (process running for 3.247)
+```
+
+When measuring the footprint of a Java application, you should measure Resident Set Size (RSS) and not the JVM heap size which is only a small part of the overall problem. The JVM not only allocates native memory for heap (-Xms, -Xmx) but also structures required by the jvm to run your application, such as compiled code and class metadata.
+
+On MacOS, to measure the RSS, run
+
+```shell
+ps aux | grep -i java | grep -i todo | grep -v grep | awk {'print $2'} | xargs ps x -o pid,rss,command -p
+```
+
+You should get something like 
+
+```shell
+  PID    RSS COMMAND
+37043 254672 java -jar target/spring-todo-app-0.0.1-SNAPSHOT.jar -Xmx600m -Xms600m
+```
+
+This shows an RSS of 254,672 KB.
+
+Instructions on measuring RSS on other platforms are available [here](https://quarkus.io/guides/performance-measure).
+
+
+### Native mode
+
+Repeat the measurement for native. Be aware, this may take some time, and there are about a hundred expected warnings. 
+
+Don't forget to clear the database before the test with `just restart-infra`.
+
+
+## Quarkus application
+
+Now we get to Quarkify the application. Quarkus's Spring compatibility libraries make the switch to Quarkus straightforward. 
+
+The simplest way to convert the pom.xml is to make a new one with `quarkus` cli, and overwrite the old one.
+
+```shell
+quarkus create
+cd code-with-quarkus 
+quarkus ext add spring-web spring-data-jpa jdbc-postgresql rest-jackson hibernate-validator
+cp pom.xml ../
+cd ..
+```
+
+You can also edit the pom by hand, which is more compelling as a demo, but harder, because it's a lot of xml to wrangle. 
+
+Whichever way you do the transformation, once the pom is correct, you remove code which is no longer needed. Delete `TodoApplication.java`, and remove all of the setup code from `TodoAppApplicationTests`. The following code should all be removed:
+
+```java
+	@Container
+	static final PostgreSQLContainer<?> DATABASE = new PostgreSQLContainer<>("postgres:15-bullseye")
+			.withDatabaseName("quarkus_test")
+			.withUsername("quarkus_test")
+			.withPassword("quarkus_test")
+			.withExposedPorts(5432);
+
+	@DynamicPropertySource
+	static void neo4jProperties(DynamicPropertyRegistry registry) {
```

Comment:
do not ask why this method is named neo4j :-)

Reply:
:)